### PR TITLE
Getting capabilities using nethermindAPI functionality

### PR DESCRIPTION
**Pull request by 1swaraj**
Fixes | Closes | Resolves #

> Anything marked as optional that you didn't need to fill in, please remove it from the PR description. Choose one of the keywords above to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666). If there is no issue, remove the line. Remove this note after reading.

## Changes:
-

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ x ] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [ x ] No

**Comments about testing , should you have some** (optional)
Will add the test cases once @kristofgazso approves the approach

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/Nethermind/Nethermind.AccountAbstraction.Test/UserOperationPoolTests.cs [hunk 1]
```diff
@@ -298,6 +301,12 @@ public void Bans_paymaster_if_it_uses_too_much_gas_for_simulation_too_many_times
             accessListSource.AccessList.Returns(new AccessList(
                 new Dictionary<Address, IReadOnlySet<UInt256>> {{new("0x0000000000000000000000000000000000000001"), new HashSet<UInt256> {0}}}));
             
+            IPeerManager peerManager = Substitute.For<IPeerManager>();
```
Comment (1swaraj) [inline src/Nethermind/Nethermind.AccountAbstraction.Test/UserOperationPoolTests.cs:304]: Consider these as placeholders for now

File: src/Nethermind/Nethermind.AccountAbstraction/Source/UserOperationPool.cs [hunk 2]
```diff
@@ -65,16 +73,33 @@ public class UserOperationPool : IUserOperationPool
...
             UserOperation userOperation = e.Key;
             SimulateAndAddToPool(userOperation, _blockTree.Head!.Header);
+            BroadcastToCompatiblePeers(userOperation, _peerManager.ConnectedPeers, _p2pProtocolHandler);
+        }
+
+        private void BroadcastToCompatiblePeers(UserOperation userOperation, IReadOnlyCollection<Peer> peers,
+            IP2PProtocolHandler protocolHandler)
+        {
+            Capability? aaCapabiltiy = new Capability(Protocol.AA, 0);
+            foreach (var peer in peers)
```
Comment (kristofgazso) [inline src/Nethermind/Nethermind.AccountAbstraction/Source/UserOperationPool.cs:96]: tip: perhaps peers.Where(peer => peer.HasAgreedCapability(aaCapability)) might be cleaner

File: src/Nethermind/Nethermind.AccountAbstraction/Source/UserOperationPool.cs [hunk 3]
```diff
@@ -65,16 +73,33 @@ public class UserOperationPool : IUserOperationPool
...
+            {
+                if (protocolHandler.HasAgreedCapability(aaCapabiltiy))
```
Comment (kristofgazso) [inline src/Nethermind/Nethermind.AccountAbstraction/Source/UserOperationPool.cs:98]: surely we want to check outside the function if our protocol can handle the capability, and only call its logic if yes

File: src/Nethermind/Nethermind.AccountAbstraction.Test/UserOperationPoolTests.cs [hunk 4]
```diff
@@ -298,6 +301,12 @@ public void Bans_paymaster_if_it_uses_too_much_gas_for_simulation_too_many_times
...
+            peerManager.MaxActivePeers.Returns(10);
+            
+            IP2PProtocolHandler p2pProtocolHandler = Substitute.For<IP2PProtocolHandler>();
```
Comment (kristofgazso) [inline src/Nethermind/Nethermind.AccountAbstraction.Test/UserOperationPoolTests.cs:307]: i recommend making a seperate test file for p2p op tests, not just expanding on the UserOperationPoolTests.cs

**Review by kristofgazso — CHANGES_REQUESTED** [review]

File: src/Nethermind/Nethermind.AccountAbstraction/Data/UserOperation.cs [hunk 5]
```diff
@@ -37,6 +38,7 @@ public UserOperation(Address target, UInt256 nonce, byte[] callData, long callGa
             Signer = signer;
             Signature = signature;
             AccessList = accessList;
+            Hash = this.CalculateHash();
```
Comment (kristofgazso) [inline src/Nethermind/Nethermind.AccountAbstraction/Data/UserOperation.cs:41]: its not going to be this easy

Comment (1swaraj) [reply src/Nethermind/Nethermind.AccountAbstraction/Data/UserOperation.cs:41]: Why so ?

Comment (1swaraj) [reply src/Nethermind/Nethermind.AccountAbstraction/Data/UserOperation.cs:41]: public static Keccak CalculateHash(this UserOperation userOperation) => Keccak.Compute(Rlp.Encode(userOperation).Bytes);
This won't work ?

Comment (kristofgazso) [reply src/Nethermind/Nethermind.AccountAbstraction/Data/UserOperation.cs:41]: bc we're going to have to come up with a rlp serialization standard for user operations and implement it. take a look at Nethermind.Mev.Data.MevBundle.GetHash

**Review by kristofgazso — APPROVED** [review]

**Pull request merged**